app.js

```diff
@@ -63,6 +63,11 @@ function parsePodcastDict(dict) {
     v = v.firstChild.nodeValue;
     o[k] = v;
     // console.log(i, dict.childNodes[i], k ,v);
+    if (k==='title') {
+      o.title = v.replace(/&/g, ' and ').replace(/\s{2,}/g, ' ');
+    } else if (k==='feedUrl') {
```

Comment:
why feedUrl need to escape?

Reply:
XML tags cannot contain unescaped ampersands.

One of the feeds in my PodcastsDB.plist included an ampersand in the query string, which got unescaped on read into JavaScript and needed to be re-escaped for the output.

Reply:
I don't think we can HTML encode URL here. What error it is when not do this?

Reply:
At least in my case, Pocket Casts on iOS failed to import OPML with unescaped ampersands. I did not get a detailed error message from the app, but it likely just failed XML validation. 

XML entities in non-CDATA contents should get replaced with their unescaped values when parsed, making it safe to use XML entities in any tag contents that are not marked as CDATA.

It would be more consistent to use the same approach (replacing single ampersands with the &amp; XML entity) for both titles and URLs, and maybe that would be better.

This is what worked for me.

Reply:
OK, I'm fine with encode here.
One more thing, could you move the replace out of `for loop` with something like

```js
if (o.hasOwnProperty("title")) {
  o.title = ....
}
```